src/plugins/data_views/common/data_views/_pattern_cache.ts

```diff
@@ -10,30 +10,50 @@ import { DataView } from './data_view';
 
 export interface DataViewCache {
   get: (id: string) => Promise<DataView> | undefined;
+  getByHash: (id: string) => DataView | undefined;
```

Comment:
to make it more generic, we could alternatively add a find function, so consumers could find matching entries. However this would be less performant of course

Reply:
Whats the benefit of the additional map?

Reply:
I was expecting id's to be calculated but that doesn't seem to be happening here. Ids are assigned as normal, when the SO is persisted. ESQL data views are looked up by hash which ignores the id. Its kind of a "I don't care which id you've came with, you're getting the one I give you!" I can imagine it works, but it certainly seems confusing if you need to trace through a data view. I'm still thinking through the implications of this.

Reply:
yes, it's an exploration, ids directly ... no hash needed ...